### PR TITLE
records: suggester enhancers refactoring

### PR DESCRIPTION
**Pull request by jmartinm**
* Aggregates all suggesters under one receiver.

* Separates each enhancement into its own function.

Signed-off-by: Javier Martin Montull <javier.martin.montull@cern.ch>

### CODE REVIEW

**Comment by jmartinm** [comment]
Continuation of https://github.com/inspirehep/inspire-next/pull/1812

@jacquerie @harunurhan have a look to see if that is a better way of organizing the suggesters.

File: inspirehep/modules/records/receivers.py [hunk 1]
```diff
@@ -391,36 +394,8 @@ def check_if_record_is_going_to_be_deleted(sender, *args, **kwargs):
...
-            if source:
-                abstract.update({
-                    'abstract_source_suggest': {
-                        'input': source,
-                        'output': source,
-                    },
-                })
+        populate_abstract_suggester(json)
+    elif 'experiments.json' in json.get('$schema'):
+        populate_experiment_name_suggester(json)
```
Comment (jacquerie) [inline inspirehep/modules/records/receivers.py]: I don't see this growing very nicely when more suggesters are added...

**Comment by jacquerie** [comment]
IMO it's a little too early to start refactoring, because we currently have two receivers with no common functionality, and in fact this refactor is just moving things around without abstracting anything.

File: inspirehep/modules/records/receivers.py (hunk 1 above)
Comment (jmartinm) [inline inspirehep/modules/records/receivers.py]: Anyway this PR was WIP to start the discussion and see if we find a better way of organizing that code, but can be closed if we don't see any improvement possible atm.

**Comment by jacquerie** [comment]
I think we agreed before Christmas to wait on this refactor until we have more examples of things we want to refactor.

**Pull request closed without merging**